src/MSBuild/TerminalLogger/TerminalLogger.cs

```diff
@@ -707,6 +716,21 @@ private void UpdateNodeStatus(BuildEventContext buildEventContext, NodeStatus? n
     /// </summary>
     private void TargetFinished(object sender, TargetFinishedEventArgs e)
     {
+        var buildEventContext = e.BuildEventContext;
```

Comment:
Comment intention here please, it's a bit too complicated to just scan and grok.